### PR TITLE
Bump supported pyfastx versions

### LINKED ISSUE #1334: Can't install with Python 3.12 on Conda
EDIT 2: This is a problem with Python 3.12 as well.

## Workarounds

- Use a managed runtime through Nextstrain CLI (following [Nextstrain install docs](https://docs.nextstrain.org/en/latest/install.html))
- Use `pip` to install augur

---

EDIT: @joverlee521 [mentioned](https://github.com/nextstrain/augur/issues/1334#issuecomment-1802497233) that pyfastx is not the only issue. Seems like an overall [issue with Python 3.11 in bioconda](https://github.com/bioconda/bioconda-recipes/issues/37805).

---

Original issue:

### Current Behavior

Currently, trying to install Augur in a Conda environment with Python 3.11 pinned does not work.

```
% micromamba install augur
…
error    libmamba Could not solve for environment specs
    The following packages are incompatible
    ├─ augur is installable with the potential options
    …
    │  ├─ augur [18.2.0|19.0.0|...|23.1.0] would require
    │  │  └─ pyfastx >=0.8.4,<0.9  with the potential options
    │  │     ├─ pyfastx 0.8.4 would require
    │  │     │  └─ python >=3.10,<3.11.0a0 , which can be installed;
    │  │     ├─ pyfastx 0.8.4 would require
    │  │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
    │  │     ├─ pyfastx 0.8.4 would require
    │  │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
    │  │     ├─ pyfastx 0.8.4 would require
    │  │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
    │  │     └─ pyfastx 0.8.4 would require
    │  │        └─ python >=3.9,<3.10.0a0 , which can be installed;
    …
    └─ pin-1 is not installable because it requires
       └─ python 3.11.* , which conflicts with any installable versions previously reported.
```

### Expected behavior

Augur should be installable with Python 3.11.

### How to reproduce

```sh
# Set up a Conda env with Python 3.11
micromamba env create -n tmp
micromamba activate tmp
# Necessary on Apple silicon:
# export CONDA_SUBDIR=osx-64
micromamba install python=3.11

mamba install augur
# Observe error
```

### Possible solution

Depend on a newer version of pyfastx. [0.8.4 is old](https://pypi.org/project/pyfastx/0.8.4/) and requires building when installed via `pip` on Python ≥3.10.

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

See commit messages.

## Related issue(s)

- Resolves #1334

## Checklist

- [x] Checks pass (I think the tests added in 1fbe47920533e01887362302f6146b34abe9cc0c should sufficiently test this change)
- [x] If making user-facing changes, add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR
- [ ] Post-merge: revert https://github.com/nextstrain/docker-base/commit/485b3e5bff48c0bb8f5909dce4cb08990917709a

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -62,7 +62,7 @@
         "packaging >=19.2",
         "pandas >=1.0.0, ==1.*",
         "phylo-treetime >=0.10.0, <0.12",
-        "pyfastx >=0.8.4, ==0.8.*",
+        "pyfastx >=0.9.0, ==0.9.*",
```
Comment (victorlin) [inline setup.py:65]: Some additional changes are necessary. [Error from CI](https://github.com/nextstrain/augur/actions/runs/6778141605/job/18423099741?pr=1335#step:8:1142):

```
E   KeyError: 'SEQ_A does not exist in fasta file'
```

Comment (victorlin) [reply setup.py:65]: Addressed with dee8a5d734468026354c26b0844662efade3a0dc...b0450991571d3536ede85f4d30d15858177bac0c

Comment (victorlin) [reply setup.py:65]: Actually, I don't think 2e7d620429eebf2fb2c721df5be20b794ca148f3 is necessary... accessing via id as noted in the pyfastx README seems to work fine. I'll have to dig a bit deeper to see what's going wrong with the tests.

Comment (joverlee521) [reply setup.py:65]: Looks like this is a [known bug in 0.9.1](https://github.com/lmdu/pyfastx/issues/62) that has been fixed in 1.0.0. 

I don't see any backwards incompatible changes in the [pyfastx changelog](https://github.com/lmdu/pyfastx/blob/ee7303d1402d5f83db1a6ec74b5140b912ae75bb/docs/changelog.rst), so I think we can bump to 1.0.* (maybe even 2.0.* ?)

Comment (victorlin) [reply setup.py]: @joverlee521 thanks for looking into this! How's a7210930b6cbd617db3c16ff3bde4889762e2f96?

Comment (joverlee521) [reply setup.py]: LGTM 👍 

I was slightly worried by the [pyfastx zlib issue](https://github.com/lmdu/pyfastx/issues/73) for docker-base, but looks like there are wheels available for Python 3.10 starting with version 0.9.0. 

Maybe we can revert https://github.com/nextstrain/docker-base/commit/485b3e5bff48c0bb8f5909dce4cb08990917709a once this is released.

**Review by joverlee521 — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (victorlin) [reply setup.py]: Good idea! Added that to the PR checklist.

**Pull request merged**